x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -511,14 +497,7 @@ private Map<String, IndicesAccessControl.IndexAccessControl> buildIndicesAccessC
             } else {
                 fieldPermissions = FieldPermissions.DEFAULT;
             }
-            indexPermissions.put(
-                index,
-                new IndicesAccessControl.IndexAccessControl(
-                    entry.getValue(),
-                    fieldPermissions,
-                    (roleQueries != null) ? DocumentPermissions.filteredBy(roleQueries) : DocumentPermissions.allowAll()
-                )
-            );
```

Comment:
If `entry.getValue()` is `false`, then `fieldPermissions` must be `FieldPermissions.DEFAULT` and `roleQueries` must be `null`.

Reply:
This is because no permission group matched the resource, or its alias/datastream, so no FLS/DLS permissions are applicable.

Reply:
> A request like GET index,alias/_search can have entry.getValue() (granted) to be false while also have non-null roleQueries.

Yeah... I though I had covered that case mentally, but you're right.

>  if a IndexAccessControl object that is not granted, the actual value of roleQueries does not seem to matter because (I think) we reject the request in all cases if there is any unauthorized names.

Yeah, I don't think any of the places where we check the `fieldPermissions` or `roleQueries` can be reached when `granted` is `false`, so that assumption is not used anywhere, but I had it on my mind and I'll need to double check that I didn't rely on it anywhere.